src/GHC/Util.hs

```diff
@@ -325,3 +330,54 @@ wToStr :: Outputable a => W a -> String
 wToStr (W e) = showPpr baseDynFlags e
 instance Outputable a => Eq (W a) where (==) a b = wToStr a == wToStr b
 instance Outputable a => Ord (W a) where compare = compare `on` wToStr
+
+-- GHC parse trees don't contain pragmas. We work around this with
+-- (nasty) parsing of comments.
+
+-- Pragmas. Comments not associated with a span in the annotations
+-- that have the form @{-# ...#-}@.
+pragmas :: ApiAnns -> [(Located AnnotationComment, String)]
+pragmas anns =
+  -- 'ApiAnns' stores pragmas in reverse order to how they were
+  -- encountered in the source file with the last at the head of the
+  -- list (makes sense when you think about it).
+  reverse
+    [ (c, s) |
+        c@(L _ (AnnBlockComment comm)) <- fromMaybe [] $ Map.lookup noSrcSpan (snd anns)
+      , let body = trimCommentDelims comm
+      , "#" `isPrefixOf` body && "#" `isSuffixOf` body
+      , let s = trim $ take (length body - 2) (drop 1 body)
+    ]
+
+-- Flags. The first element of the pair is the (located) annotation
+-- comment that sets the flags enumerated in the second element of the
+-- pair.
+flags :: ApiAnns -> [(Located AnnotationComment, [String])]
+flags anns =
+  [(c, opts) | (c, s) <- pragmas anns, "OPTIONS_GHC" `isPrefixOf` s
+    , let opts = words $ fromJust (stripPrefix "OPTIONS_GHC" s)]
+  ++
+  -- Old versions of GHC accepted 'OPTIONS' rather than 'OPTIONS_GHC' (but
+  -- this is deprecated).
+  [(c, opts) | (c, s) <- pragmas anns, "OPTIONS" `isPrefixOf` s && not ("OPTIONS_" `isPrefixOf` s)
```

Comment:
Sceptical there isn't a more concise way to express this... How about:

    stripPrefix "OPTIONS_GHC" s `mplus` stripPrefix "OPTIONS" s

Reply:
Each one of these exercises teaches me something new 😄 Will look into it!

Reply:
✔️